whipper/common/encode.py

```diff
@@ -89,3 +89,62 @@ def _tag(self):
         w.save()
 
         self.stop()
+
+
+class EmbedPictureTask(task.Task):
+    description = 'Embed picture to FLAC'
+
+    def __init__(self, track_path, coverart_path):
+        self.track_path = track_path
+        self.coverart_path = coverart_path
+
+    def start(self, runner):
+        task.Task.start(self, runner)
+        self.schedule(0.0, self._embed_picture)
+
+    def _make_flac_picture(self, coverart_filename):
+        """Given a path to a jpeg or png file, return a flac.Picture
+
+        Returns: a valid flac.Picture or None
+        """
+        if not coverart_filename:
+            return None
+
+        try:
+            from PIL import Image
+        except ImportError as e:
+            logger.warning("pillow is required to grab cover art: %r", e)
+
+        im = Image.open(coverart_filename)
+        if im.format == 'JPEG':
+            mime = 'image/jpeg'
+        elif im.format == 'PNG':
+            mime = 'image/png'
+        else:
+            return None          # we only support png and jpeg
+
+        pic = Picture()
+        with open(coverart_filename, 'rb') as f:
+            pic.data = f.read()
+
+        pic.type = 3             # id3.PictureType.COVER_FRONT
+        pic.mime = mime
+        pic.width, pic.height = im.size
+        if im.mode == 'P':       # 8-bit pixels, mapped using color palette
```

Comment:
Should we check modes?

Reply:
Since its saved as jpg anyway so i think no.

Reply:
I don't think that the file extension matters, it's its content to define which format it really is (as long as `Image.open()` can handle it).

Reply:
> Should we check modes at all?

I'm not sure about this, @MerlijnWajer , what do you think ?.

Reply:
Asking @Freso too.

Reply:
It doesn’t seem like the mode is relevant to anything here. Either we save the image as-is—in which case it’s saved as the mode it already is—or we convert it to whatever we want (it seems JPEG here?)—in which case the software doing the conversion is hopefully smarter about handling the mode than we could or should be here.

If only a number of modes are supported for embedding at all, just testing whether it’s one of those (`if not im.mode in ('P', 'RGB', 'SRGB'): return`) should be enough.